api/internal/controllers/analysis/analysis.go

```diff
@@ -147,11 +159,88 @@ func (c *Controller) hasDuplicatedHash(
 }
 
 func (c *Controller) publishInBroker(analysisID uuid.UUID) error {
-	response, err := c.GetAnalysis(analysisID)
+	responsee, err := c.GetAnalysis(analysisID)
 	if err != nil {
 		return err
 	}
 
 	return c.broker.Publish("", exchange.NewAnalysis,
-		exchange.Fanout, response.ToBytes())
+		exchange.Fanout, responsee.ToBytes())
+}
+
+//TODO:REMOVE ALL BELOW AFTER v2.10.0
+func (c *Controller) treatCompatibility(rawAnalysis *analysis.Analysis) error {
+	if !c.existsDepecratedHashesSlice(rawAnalysis.AnalysisVulnerabilities) {
+		return nil
+	}
+	hashSlice := make([]string, 0)
+
+	for i := range rawAnalysis.AnalysisVulnerabilities {
+		hashSlice = append(hashSlice,
+			rawAnalysis.AnalysisVulnerabilities[i].Vulnerability.DeprecatedHashes...)
+	}
+
+	if err := c.saveUpdates(hashSlice, rawAnalysis); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Controller) saveUpdates(hashSlice []string, rawAnalysis *analysis.Analysis) error {
+	res := c.repoAnalysis.FindVulnerabilitiesByHashSliceInRepository(hashSlice, rawAnalysis.RepositoryID)
+	if res.GetError() != nil {
+		return res.GetError()
+	}
+	mapHashToID, err := c.parseResIds(res)
+	if err != nil {
+		return err
+	}
+	query := c.mountUpdateQuery(rawAnalysis, mapHashToID)
+
+	if err := c.repoAnalysis.RawQuery(query); err != nil {
+		return err
+	}
+	return nil
+}
+
+/* mountUpdateQuery iterates over rawAnalysis.AnalysisVulnerabilities and
+checks if some vuln.Vulnerability.DeprecatedHashes is present on
+mapHashToId then creates and update statement to update the
+deprecated Hash value to the new one (that is present in rawAnalysis.Vulnerability.VulnHash field) */
```

Comment:
Use the same pattern of comments used in other places:
Replace `/* */` for single line comments using `//`